### DISCUSSION TITLE
expect assertion fails when filtered by table name

### DISCUSSION THREAD

**Discussion by InduKrish** [testing-library/playwright-testing-library#529] [Q&A] [answered]
I would like to click the checkbox in the corresponding row after the text that i want to search is found in the table (question is addressed in #517)

when i tried meanwhile by table name to test this way , i see the following error, although i see in the output DOM, that the element is found. I tried using both getByText and FindByText(), and both failed.

        const header = await this.screen.findByTestId("editable-nested-table-GroupCode");
        const base =   await this.within(header).findByText(" LINEHOLDER")
        console.log("value " + base);
        await expect(base).toContainText(text);

output:

<img width="1775" alt="Screen Shot 2022-09-20 at 11 37 01 AM" src="https://user-images.githubusercontent.com/113629123/191315656-819e34f0-68da-476d-8f4b-7db7704bd9dd.png">
<img width="1775" alt="Screen Shot 2022-09-20 at 11 37 48 AM" src="https://user-images.githubusercontent.com/113629123/191315677-25281cb5-0084-4df5-9226-a5424c1b6b81.png">
<img width="1775" alt="Screen Shot 2022-09-20 at 11 38 22 AM" src="https://user-images.githubusercontent.com/113629123/191315700-ddd7a30e-85b7-47a5-9ea4-42a02bb4024a.png">

**Code that works:** while filtering by row id, however i want to filter by table id.

 Assertion for  row test id erow-GroupCode-0 works.
 const header = await this.screen.findByTestId('erow-GroupCode-0');
        console.log(" header" + header)
        const base = this.within(header).getByText("LINEHOLDER")
        console.log("base value" + base);
        const check = this.within(header).getByText("check")
        console.log(" value " + check)
        await expect(base).toContainText(text);

output console log:
Mac-C02FT4MH:cbna-test-integration-cbna x273092$ npx playwright test

Running 1 test using 1 worker
[chromium] › tests/web/vacation.bidaward.defaults.spec.po.new.js:14:9 › Bid and award Default page verification › Default page verification
 headerLocator@query-by-test-id=["erow-GroupCode-0"]
base valueLocator@query-by-test-id=["erow-GroupCode-0"] >> get-by-text=["LINEHOLDER"]
 value Locator@query-by-test-id=["erow-GroupCode-0"] >> get-by-text=["check"]

<img width="1792" alt="Screen Shot 2022-09-20 at 10 48 25 AM" src="https://user-images.githubusercontent.com/113629123/191305168-793854d9-ad60-4187-a0e4-aa7a16a19cfd.png">

html structure:
<img width="1262" alt="Screen Shot 2022-09-20 at 11 43 01 AM" src="https://user-images.githubusercontent.com/113629123/191316324-f1a3f682-5f28-43a5-b36f-8c69469aab08.png">

why assertion is working while filtering by row test id erow-GroupCode-0, but it fails while filtering by table name -editable-nested-table-GroupCode?

**Comment by jrolfs** [comment, marked as answer]
Your text match is probably too specific when matching against a parent element. See https://testing-library.com/docs/queries/about#textmatch. Please try to reference the Testing Library documentation when troubleshooting queries.

```ts
const header = await this.screen.findByTestId("editable-nested-table-GroupCode");
const base =   await this.within(header).findByText(/LINEHOLDER/); // <--- Use a `RegExp` here?

await expect(base).toContainText(text);
```

**Reply by jrolfs** [reply]
@InduKrish let me know if this works for you.

**Reply by InduKrish** [reply]
> Your text match is probably too specific when matching against a parent element. See https://testing-library.com/docs/queries/about#textmatch. Please try to reference the Testing Library documentation when troubleshooting queries.
> 
> ```ts
> const header = await this.screen.findByTestId("editable-nested-table-GroupCode");
> const base =   await this.within(header).findByText(/LINEHOLDER/); // <--- Use a `RegExp` here?
> 
> await expect(base).toContainText(text);
> ```


After adding RegExp , the above snippet is working fine! Thank you for your help! Much appreciated!
